File: 2487-remove-nodes-from-linked-list/2487-remove-nodes-from-linked-list-stack.java
class Solution {
    public ListNode removeNodes(ListNode head) {
        Stack<ListNode> stack = new Stack<>();

        while (head != null) {
            stack.push(head);
            head = head.next;
        }

        ListNode currentNode = stack.pop();

        while (!stack.isEmpty()) {

            ListNode node = stack.pop();

            if (node.val >= currentNode.val) {
                node.next = currentNode;
                currentNode = node;
            }
        }

        return currentNode;
    }
}

File: 2487-remove-nodes-from-linked-list/2487-remove-nodes-from-linked-list-stack.py
class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:

        stack = []

        while head:
            stack.append(head)
            head = head.next

        current_node = stack.pop()

        while stack:

            node = stack.pop()

            if node.val >= current_node.val:
                node.next = current_node
                current_node = node

        return current_node
